info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -0,0 +1,268 @@
+    }
+    Query<Entity> electionQuery = Query.newEntityQueryBuilder()
+        .setKind("Election")
+        .setFilter(PropertyFilter.eq("queryId", electionQueryId))
+        .build();
+    QueryResults<Entity> electionQueryResults = datastore.run(electionQuery);
+    Entity electionEntity = electionQueryResults.next();
+    // Create a copy since the original list is a
+    // {@code com.google.common.collect.ImmutableCollection}.
+    List<Value<String>> candidatePositions =
+        new ArrayList<>(electionEntity.getList("candidatePositions"));
+    List<Value<String>> candidateIds =
+        new ArrayList<>(electionEntity.getList("candidateIds"));
+    List<Value<Boolean>> candidateIncumbency =
+        new ArrayList<>(electionEntity.getList("candidateIncumbency"));
+    // Obtain candidate information and create candidate entities in the database.
+    for (JsonElement candidate : candidates) {
+      storeElectionContestCandidateInDatabase((JsonObject) candidate,
+          candidateIds, candidateIncumbency);
+      candidatePositions.add(
+          StringValue.newBuilder(
+              contest.get("office").getAsString()).build());
+    }
+    // Fill in position and candidate information for the election entities in the database.
+    electionEntity = Entity.newBuilder(electionEntity)
+        .set("candidatePositions", candidatePositions)
+        .build();
+    electionEntity = Entity.newBuilder(electionEntity)
+        .set("candidateIds", candidateIds)
+        .build();
+    electionEntity = Entity.newBuilder(electionEntity)
+        .set("candidateIncumbency", candidateIncumbency)
+        .build();
+    datastore.update(electionEntity);
+  }
+
+  // @TODO [Find incumbency.]
+  /**
+   * Stores information of a {@code candidate} in the database, and updates the candidate's
+   * running position's information for the election.
+   * Information includes: name, party affiliation.
+   */
+  private void storeElectionContestCandidateInDatabase(JsonObject candidate,
+      List<Value<String>> candidateIds, List<Value<Boolean>> candidateIncumbency) {
+    String name = candidate.get("name").getAsString();
+    String party = candidate.get("party").getAsString();
+    // @TODO [May expand to other information to uniquely identify a candidate. Currently,
+    // candidate information includes only name and party affiliation.]
+    long candidateId = (long) (name.hashCode() + party.hashCode());
+    Key candidateKey = datastore.newKeyFactory()
+        .setKind("Candidate")
+        .newKey(candidateId);
+    Entity candidateEntity = Entity.newBuilder(candidateKey)
+        .set("name", name)
+        .set("partyAffiliation", party + " Party")
+        .build();
+    datastore.put(candidateEntity);
+    candidateIds.add(StringValue.newBuilder(Long.toString(candidateId)).build());
+    candidateIncumbency.add(BooleanValue.newBuilder(false).build());
+  }
+
+  // For testing purposes.
+  public static void main(String[] args) {
```

Comment:
When you do go back and add unit tests, I'd remove any unnecessary main methods that are only there for testing.

Reply:
Will do!